client/my-sites/domains/domain-management/components/domain/transfer-flag.jsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import PropTypes from 'prop-types';
+import React, { PureComponent } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { type as domainTypes, transferStatus } from 'lib/domains/constants';
+import { localize } from 'i18n-calypso';
+import Notice from 'components/notice';
+
+class DomainTransferFlag extends PureComponent {
+	static propTypes = {
+		domain: PropTypes.object.isRequired,
+		translate: PropTypes.func.isRequired,
+	};
+
+	render() {
+		const { domain, translate } = this.props;
+
+		if ( domain.type !== domainTypes.TRANSFER ) {
+			return null;
+		}
+
+		let status = 'is-warning';
```

Comment:
Why not put this in the `default` section instead?

Reply:
Not sure what you mean? Because we do need the status for `case transferStatus.PENDING_OWNER:`.

Reply:
Sorry, should have been more clear: I meant it like this:
```
let status, message;
switch ( domain.transferStatus ) {
    case transferStatus.PENDING_OWNER:
        status = x;
        message = '';
        break;

    case transferStatus.CANCELLED:
        status = x;
        message = '';
        break;

    default:
        status = x;
        message = '';
        break;
```

Your intent is much more clearer this way, IMHO. I understand that you wanted to "re-use"/share the status with `transferStatus.PENDING_OWNER` case but I don't think that's worth the increased complexity of the code 😅